src/ClassComponent/LastPressedKey2/LastPressedKey2.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+
+type State = {
+  key: string;
+};
+
+export class LastPressedKey2 extends React.PureComponent<{}, State> {
+  state: State = {
+    key: '',
+  };
+
+  componentDidMount(): void {
+    document.addEventListener('keyup', this.handleKeyPress);
+  }
+
+  componentWillUnmount(): void {
+    document.removeEventListener('keyup', this.handleKeyPress);
+    // console.log('componentWillUnmount LastPressedKey2');
+  }
+
+  handleKeyPress = (event: KeyboardEvent) => {
+    this.setState({ key: event.key });
+  };
+
+  render() {
+    const { key } = this.state;
+
+    return (
+      <div>
+        {!key ? (
+          <p>Nothing was pressed yet</p>
+        ) : (
+          <p className="App__message">{`The last pressed key is [${key}]`}</p>
+        )}
```

Comment:
```suggestion
          <p className="App__message">
            {key ? `The last pressed key is [${key}]` : 'Nothing was pressed yet'}
          </p>
```
for fix tests logic should be in class component.
Use this component in App or move logic from this component to App component